app/src/main/java/com/onboarding/pronosticAR/data/mapper/WeatherMapper.kt

```diff
@@ -0,0 +1,35 @@
+package com.onboarding.pronosticAR.data.mapper
+
+import com.onboarding.pronosticAR.data.response.DescriptionResponse
+import com.onboarding.pronosticAR.data.response.TemperatureResponse
+import com.onboarding.pronosticAR.data.response.WeatherApiResponse
+import com.onboarding.pronosticAR.data.response.WeatherForecastResponse
+import com.onboarding.pronosticAR.mvp.entity.Weather
+import com.onboarding.pronosticAR.mvp.entity.WeatherDescription
+import com.onboarding.pronosticAR.mvp.entity.WeatherForecast
+import com.onboarding.pronosticAR.mvp.entity.WeatherTemperature
+
+fun WeatherApiResponse.transform(): Weather {
+    return Weather(this.list.transform())
+}
+
+fun List<WeatherForecastResponse>.transform(): List<WeatherForecast> {
+    val list = mutableListOf<WeatherForecast>()
+    this.forEach {
+        list.add(it.transform())
+    }
+    return list
+}
```

Comment:
here you can simplify this using .map {
https://grokonez.com/kotlin/kotlin-transform-list-map-methods-example

Reply:
done